pypeman/contrib/http.py

```diff
@@ -154,12 +158,19 @@ async def handle_request(self, request):
         msg = message.Message(content_type='http_request', payload=content, meta=meta)
         try:
             result = await self.handle(msg)
-            encoding = self.encoding or 'utf-8'
-            return web.Response(
-                body=str(result.payload).encode(encoding),
-                status=result.meta.get('status', 200),
-                content_type=getattr(result, "content_type", None),
-            )
+            if self.binary_response:
+                return web.Response(
+                    body=result.payload,
```

Comment:
please adding description to merge requests.

So if I understand correctly you want to tell a node, that the responses it will receive will be binary.
Shouldn't / Couldn't the type (text / binary) be automatically derived from the content type of the http response?


What about links that could return text or binary? (download from a GED for example)
same question about encoding.

Why do we specify the encoding of a response?

SHouldn't a response tell us what encoding we get?


In other words.

Shouldn't a node have three options.

text / binary and autodetect

and for encoding: autodetect if None otherwise the encoding passed to the node?

Reply:
Yup, desc added and, change to auto set type (removed the param), and for encoding, i don't know if it's a good idea to have "random" encodings, the dev cannot be sure of what is sent and i think it's a problem